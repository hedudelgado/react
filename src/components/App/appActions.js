import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
