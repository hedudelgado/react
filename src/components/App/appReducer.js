import {
  SET_NOTIFICATION_MESSAGE, SET_EDIT_VIEW, DEFAULT_VIEW,
} from './appActions';

const defaultState = {
  editViewName: DEFAULT_VIEW,
};

export default (state = defaultState, action) => {
  switch (action.type) {
    case SET_EDIT_VIEW:
      return {
        ...state,
        editViewName: action.editViewName,
      };
    case SET_NOTIFICATION_MESSAGE:
      return {
        ...state,
        notificationMessage: action.notificationMessage,
      };
    default:
      return state;
  }
};
