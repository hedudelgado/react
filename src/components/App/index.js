export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
