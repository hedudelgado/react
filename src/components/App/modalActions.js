export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';

export const openModal = content => ({
  type: MODAL_OPEN,
  content,
});

export const closeModal = () => ({
  type: MODAL_CLOSE,
});
