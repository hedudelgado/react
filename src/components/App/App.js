import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
