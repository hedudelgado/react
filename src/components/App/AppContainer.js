import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
