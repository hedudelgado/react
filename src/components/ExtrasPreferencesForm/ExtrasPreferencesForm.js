import React from 'react';
import {
  func,
  object,
} from 'prop-types';
import Button from '../../layout/Button';
import Typography from '../../layout/Typography';
import { DEFAULT_VIEW } from '../App/appActions';
import RadioGroup, { RadioItem } from '../../layout/RadioGroup';
import { mealOptions } from '../../config.json';

const ExtrasPreferencesForm = ({
  profile: {
    business,
  },
  dictionary,
  setEditView,
  values,
  handleChange,
  handleBlur,
  handleSubmit,
}) => (
  <form
    data-test="extras-preferences-form"
    name="extras-preferences-form"
    onSubmit={handleSubmit}
  >
    <Typography component="h4" className="pi-color-greyscale-darkest">
      {dictionary['extraspreferences.edit.meals.title']}
    </Typography>
    <RadioGroup>
      {
        mealOptions.map(({ code, dictionaryKey }) => (
          <RadioItem
            field={{
              name: 'mealOptions',
              value: values.mealOptions,
              onBlur: handleBlur,
              onChange: handleChange,
            }}
            label={dictionary[dictionaryKey]}
            name="mealOptions"
            id={code}
            value={code}
            key={code}
          />
        ))
      }
    </RadioGroup>
    {business && (
      <div data-test="extraspreferences-wifi">
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['extraspreferences.edit.wifi.title']}
        </Typography>
        <RadioGroup>
          <RadioItem
            field={{
              name: 'wifiOptions',
              value: values.wifiOptions,
              onBlur: handleBlur,
              onChange: handleChange,
            }}
            label={dictionary['extraspreferences.edit.wifi.description.allow']}
            name="wifiOptions"
            id="allowWifi"
            value="true"
          />
          <RadioItem
            field={{
              name: 'wifiOptions',
              value: values.wifiOptions,
              onBlur: handleBlur,
              onChange: handleChange,
            }}
            label={dictionary['extraspreferences.edit.wifi.description.never']}
            name="wifiOptions"
            id="noWifi"
            value="false"
          />
        </RadioGroup>
      </div>
    )}
    <Typography component="h4" className="pi-color-greyscale-darkest">
      {dictionary['extraspreferences.edit.invoice.title']}
    </Typography>
    <RadioGroup>
      <RadioItem
        field={{
          name: 'invoiceOptions',
          value: values.invoiceOptions,
          onBlur: handleBlur,
          onChange: handleChange,
        }}
        label={dictionary['extraspreferences.edit.invoice.email']}
        name="invoiceOptions"
        id="emailInvoice"
        value="true"
      />
      <RadioItem
        field={{
          name: 'invoiceOptions',
          value: values.invoiceOptions,
          onBlur: handleBlur,
          onChange: handleChange,
        }}
        label={dictionary['extraspreferences.edit.invoice.checkin']}
        name="invoiceOptions"
        id="checkInInvoice"
        value="false"
      />
    </RadioGroup>
    <Button
      data-test="save-changes"
      className="wb-push--bottom"
      color="primary"
      type="submit"
    >
      {dictionary['extraspreferences.button.save']}
    </Button>
    <Button
      data-test="closeForm"
      color="default"
      variant="text"
      className="underline block"
      fullWidth
      onClick={() => setEditView(DEFAULT_VIEW)}
    >
      {dictionary['extraspreferences.button.cancel']}
    </Button>
  </form>
);


ExtrasPreferencesForm.propTypes = {
  profile: object.isRequired,
  dictionary: object.isRequired,
  values: object.isRequired,
  setEditView: func.isRequired,
  handleChange: func.isRequired,
  handleBlur: func.isRequired,
  handleSubmit: func.isRequired,
};

export default ExtrasPreferencesForm;
