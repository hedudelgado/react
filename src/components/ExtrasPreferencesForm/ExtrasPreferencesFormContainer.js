import React, { PureComponent } from 'react';
import { func, object } from 'prop-types';
import { Formik } from 'formik';
import BaseStyle from '../../layout/BaseStyle';
import Typography from '../../layout/Typography';
import ExtrasPreferencesForm from './ExtrasPreferencesForm';
import {
  DEFAULT_VIEW,
  EXTRAS_PREFERENCES_ERROR,
  EXTRAS_PREFERENCES_VIEW,
} from '../App/appActions';
import Notification from '../../layout/Notification';

export default class ExtrasPreferencesFormContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(values, actions) {
    const {
      setNotificationMessage,
      updateProfileStore,
      setEditView,
      updateProfile,
    } = this.props;

    const payload = this.createPayload(values);

    updateProfile(payload)
      .then(() => {
        setNotificationMessage(EXTRAS_PREFERENCES_VIEW);
        setEditView(DEFAULT_VIEW);
        actions.setSubmitting(false);
        updateProfileStore(payload);
      }).catch(() => {
        setNotificationMessage(EXTRAS_PREFERENCES_ERROR);
      });
  }

  createPayload(values) {
    const { profile } = this.props;
    const payload = {
      ...profile,
      bookingPreference: {
        ...profile.bookingPreference,
        foodPreference: +values.mealOptions,
        preselectWifi: values.wifiOptions === 'true',
      },
      paymentPreference: {
        ...profile.paymentPreference,
        electronicInvoiceRequired: values.invoiceOptions === 'true',
      },
    };

    return payload;
  }

  render() {
    const {
      app: { notificationMessage },
      dictionary,
      setEditView,
      profile,
    } = this.props;

    const {
      bookingPreference: {
        foodPreference = 0,
        preselectWifi = false,
      } = {},
      paymentPreference: {
        electronicInvoiceRequired = false,
      } = {},
    } = profile;

    return (
      <BaseStyle component="div" className="component-container">
        {notificationMessage === EXTRAS_PREFERENCES_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['extraspreferences.edit.title']}
        </Typography>
        <Typography component="p" mb="3">
          {dictionary['extraspreferences.edit.description']}
        </Typography>

        <Formik
          data-test="extras-preferences-form-wrapper"
          onSubmit={this.handleSubmit}
          initialValues={{
            mealOptions: `${foodPreference}`,
            wifiOptions: `${preselectWifi}`,
            invoiceOptions: `${electronicInvoiceRequired}`,
          }}
          render={formikProps => (
            <ExtrasPreferencesForm
              {...formikProps}
              dictionary={dictionary}
              setEditView={setEditView}
              profile={profile}
            />
          )
          }
        />
      </BaseStyle>
    );
  }
}

ExtrasPreferencesFormContainer.propTypes = {
  app: object.isRequired,
  profile: object.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  setNotificationMessage: func.isRequired,
  updateProfileStore: func.isRequired,
  updateProfile: func.isRequired,
};
