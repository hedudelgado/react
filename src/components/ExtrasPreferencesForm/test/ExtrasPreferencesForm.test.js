import React from 'react';
import { shallow } from 'enzyme';
import ExtrasPreferencesForm from '../ExtrasPreferencesForm';
import { DEFAULT_VIEW } from '../../App/appActions';

describe('ExtrasPreferencesForm Component', () => {
  let wrapper;

  const props = {
    profile: {
      sessionId: 'zdiKE1MflNUmdXAS',
      contactDetail: {
        title: 'Mrs',
        firstName: 'Test',
        lastName: 'Test',
        email: 'test@whitbread.com',
        telephone: '111111111',
        mobile: '+351222222222',
        nationality: 'GB',
        passport: {
          number: '',
          countryOfIssue: '',
        },
        carRegistration: '',
        address: {
          line1: '120 Holborn',
          line2: '',
          line3: '',
          line4: 'LONDON',
          line5: '',
          postCode: 'EC1N 2TD',
          countryCode: 'GB',
          type: 'BUSINESS',
          companyName: 'Whitbread',
        },
      },
      paymentPreference: {
        electronicInvoiceRequired: true,
        paymentCard: {
          cardType: 'VI',
          cardNumber: '************1111',
          expiryDate: '07/23',
          cardHolderName: 'dwa',
        },
      },
      additionalGuests: [
        {
          title: 'Mr',
          firstName: 'Mac',
          lastName: 'Donalds',
          email: 'mac@test.com',
          telephone: '1111111111',
          mobile: '+441111111111',
          nationality: 'RP',
          carRegistration: '',
        },
      ],
      bookingPreference: {
        roomRequirements: {
          type: 'DB',
          adults: 2,
          children: 0,
          cotRequired: false,
          hotelBrand: 'PI',
        },
        foodPreference: 11,
        wantSmsConfirmations: false,
      },
      companyName: 'Pareto Law',
      companyId: '',
      businessUse: false,
      guestHistoryNumber: 'G14187551',
    },
    dictionary: {
      'extraspreferences.button.cancel': 'Cancel changes',
      'extraspreferences.button.save': 'Save changes',
      'mealoptions.pibreakfast': 'Premier Inn Breakfast',
      'mealoptions.continentalbreakfast': 'Continental Breakfast',
      'mealoptions.mealdeal': 'Meal Deal',
      'mealoptions.nomeals': 'No Meals',
    },
    values: {
      mealOptions: '17',
    },
    setEditView: jest.fn(),
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<ExtrasPreferencesForm {...props} />);
  });

  describe('Renders ExtrasPreferencesForm Component Elements', () => {
    it('should render the component', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should return to Extras Preferences default view when changes are cancelled', () => {
      wrapper.find('[data-test="closeForm"]').simulate('click');
      expect(props.setEditView).toHaveBeenCalledWith(DEFAULT_VIEW);
    });

    it('should not show extras wifi preferences if I am a leisure customer', () => {
      expect(wrapper.find('[data-test="extraspreferences-wifi"]')).toHaveLength(0);
    });

    it('should show the extras wifi preferences if I am a business customer', () => {
      const newProps = {
        ...props,
        profile: {
          ...props.profile,
          business: {
            accessLevel: 'SUPER',
          },
        },
      };
      wrapper = shallow(<ExtrasPreferencesForm {...newProps} />);
      expect(wrapper.find('[data-test="extraspreferences-wifi"]')).toHaveLength(1);
    });
  });
});
