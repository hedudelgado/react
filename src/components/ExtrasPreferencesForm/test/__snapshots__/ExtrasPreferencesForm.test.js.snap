// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtrasPreferencesForm Component Renders ExtrasPreferencesForm Component Elements should render the component 1`] = `
<form
  data-test="extras-preferences-form"
  name="extras-preferences-form"
  onSubmit={[MockFunction]}
>
  <Typography
    className="pi-color-greyscale-darkest"
    component="h4"
  />
  <RadioGroup>
    <RadioItem
      field={
        Object {
          "name": "mealOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "17",
        }
      }
      id="11"
      key="11"
      label="Premier Inn Breakfast"
      name="mealOptions"
      subLabels={Array []}
      value="11"
    />
    <RadioItem
      field={
        Object {
          "name": "mealOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "17",
        }
      }
      id="12"
      key="12"
      label="Continental Breakfast"
      name="mealOptions"
      subLabels={Array []}
      value="12"
    />
    <RadioItem
      field={
        Object {
          "name": "mealOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "17",
        }
      }
      id="17"
      key="17"
      label="Meal Deal"
      name="mealOptions"
      subLabels={Array []}
      value="17"
    />
    <RadioItem
      field={
        Object {
          "name": "mealOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": "17",
        }
      }
      id="0"
      key="0"
      label="No Meals"
      name="mealOptions"
      subLabels={Array []}
      value="0"
    />
  </RadioGroup>
  <Typography
    className="pi-color-greyscale-darkest"
    component="h4"
  />
  <RadioGroup>
    <RadioItem
      field={
        Object {
          "name": "invoiceOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": undefined,
        }
      }
      id="emailInvoice"
      name="invoiceOptions"
      subLabels={Array []}
      value="true"
    />
    <RadioItem
      field={
        Object {
          "name": "invoiceOptions",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": undefined,
        }
      }
      id="checkInInvoice"
      name="invoiceOptions"
      subLabels={Array []}
      value="false"
    />
  </RadioGroup>
  <Button
    className="wb-push--bottom"
    color="primary"
    data-test="save-changes"
    fullWidth={false}
    href={null}
    onClick={[Function]}
    submit={false}
    type="submit"
    variant="button"
  >
    Save changes
  </Button>
  <Button
    className="underline block"
    color="default"
    data-test="closeForm"
    fullWidth={true}
    href={null}
    onClick={[Function]}
    submit={false}
    variant="text"
  >
    Cancel changes
  </Button>
</form>
`;
